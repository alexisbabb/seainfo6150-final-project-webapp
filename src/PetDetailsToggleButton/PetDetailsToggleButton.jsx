import React from "react";
import PropTypes from "prop-types";
import styles from "./PetDetailsToggleButton.module.css";

const PetDetailsToggleButton = (props) => {
  return (
    <button className={styles.detailsButton} onClick={props.onClick}>
      {props.buttonText}
    </button>
  );
};

PetDetailsToggleButton.propTypes = {
  buttonText: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};
export default PetDetailsToggleButton; 