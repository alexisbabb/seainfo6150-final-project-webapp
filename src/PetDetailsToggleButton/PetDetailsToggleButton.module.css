.detailsButton {
    background: #b8610f;
    border-radius: 1em;
    border-color: rgba(255, 255, 255, 0); 
    color: #ffffff;
    padding: 1em;
    width: 75%; 
    display: block; 
    margin-left: auto; 
    margin-right: auto; 
}