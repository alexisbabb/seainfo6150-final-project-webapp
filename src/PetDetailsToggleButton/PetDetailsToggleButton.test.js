import React from "react";
import PetDetailsToggleButton from "./PetDetailsToggleButton"; 

describe("PetDetailsToggleButton tests", () => {

    let isTextShowing = false;

	function onClick() {
		setIsTextShowing(!isTextShowing);
	}

    it("renders correctly", () => {
        const buttonText = "Show More"; 
        const { container } = render(<PetDetailsToggleButton buttonText={buttonText} onClick={onClick}/>);
        expect(container).toMatchSnapshot();
    });
});