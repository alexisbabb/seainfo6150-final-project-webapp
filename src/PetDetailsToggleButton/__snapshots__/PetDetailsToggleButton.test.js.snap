// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetDetailsToggleButton tests renders correctly 1`] = `
<div>
  <button
    class="detailsButton"
  >
    Show More
  </button>
</div>
`;
